code/parse/sexp.cpp

```diff
@@ -4783,7 +4783,9 @@ void stuff_sexp_text_string(SCP_string &dest, int node, int mode)
 			{
 				// if (for whatever reason) we have an empty string or an invalid number, print 0
 				// do not do this trick in error mode, because we want to know what the text actually is
+				// note that we let <argument> strings pass here
 				if ((mode != SEXP_ERROR_CHECK_MODE) &&
+					(stricmp(ctext_string, "<argument>")) &&
```

Comment:
This should check `Sexp_nodes[node].flags & SNF_SPECIAL_ARG_IN_NODE` per the recent reduction of sexp string comparisons.